pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1945,8 +1945,12 @@ public boolean getBoolean(int columnIndex) throws SQLException {
       return false; // SQL NULL
     }
 
+    int col = columnIndex - 1;
+    if (Oid.BOOL == fields[col].getOID()) {
+      return 't' == (char) (this_row[col][0] & 0xFF);
```

Comment:
Does this buy much?
Can a test be added to cover that?
Is this resilient against ArrayIndexOutOfBoundsException?